### PR TITLE
🎉Source WooCommerce: migrate to low-code

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolving #18707

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: docs/integrations/sources/woocommerce.md
Comment (misteryeo) [inline docs/integrations/sources/woocommerce.md:49]: Shouldn't this say WooCommerce?

File: docs/integrations/sources/woocommerce.md
Comment (misteryeo) [inline docs/integrations/sources/woocommerce.md:23]: Shouldn't this be WooCommerce?

**Review by misteryeo — DISMISSED** [review]

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml
Comment (girarda) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml:11]: I don't think this line does anything. do we need a custom requster?

**Review by girarda — DISMISSED** [review]

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml]: fixed, extra line

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml:273]: nit extra space

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml
Comment (brianjlai) [inline airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml:18]: In the original python code, we were appending user agent and format headers. We have a field `request_headers` to allow for that. Are they no longer required to make requests to the API?

**Review by brianjlai — COMMENTED** [review]
Some feedback, formatting, and questions about the .yaml implementation. Have you pulled the latest master recently? Looking over some of the failing tests are unrelated to your changes, but they have been fixed in the past few days.

Also, some of your schema and spec files appear to be missing newlines, can you run linting/formatting or ensure your pre-commit hooks are run before merging your code in

File: docs/integrations/sources/woocommerce.md
Comment (artem1205) [reply docs/integrations/sources/woocommerce.md]: fixed

File: docs/integrations/sources/woocommerce.md
Comment (artem1205) [reply docs/integrations/sources/woocommerce.md]: fixed

File: airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml
Comment (artem1205) [reply airbyte-integrations/connectors/source-woocommerce/source_woocommerce/woocommerce.yaml:246]: fixed

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml
Comment (brianjlai) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:1819]: I'm seeing a part of the CI build phase failing with the following error:
```
LocalDefinitionsProviderTest > initializationError FAILED
    java.lang.UnsupportedOperationException: There is no seed spec for docker image airbyte/source-woocommerce:0.2.0
```
Since we haven't published the image yet and it can't be found, is this an expected failure? I see on some PRs you've bump it, but other not.

**Review by brianjlai — COMMENTED** [review]
The new yaml manifest looks good. We should still try to ensure we have a green build before publishing and merging the code. The tests still look to be unrelated to yours, but lets try pulling the latest code one more time to see if the connector builder server failing unit tests start passing. After that I will approve

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml
Comment (artem1205) [reply airbyte-config/init/src/main/resources/seed/source_definitions.yaml:1819]: not sure about this exact test, but I guess it is ok for other unittests to fail, as this PR is still blocked by https://github.com/airbytehq/airbyte/issues/19753

**Review by brianjlai — APPROVED** [review]
Integration and low code looks great. Thanks for incorporating the feedback. I also noticed that some of your schema files are being flagged as missing newline chars at the end of the file. Can you please add them in wherever they are missing.

**Comment by artem1205** [comment]
> Integration and low code looks great. Thanks for incorporating the feedback. I also noticed that some of your schema files are being flagged as missing newline chars at the end of the file. Can you please add them in wherever they are missing.

Fixed

**Pull request merged**
